corgidrp/fluxcal.py

```diff
@@ -0,0 +1,229 @@
+# This module is written to do an absolute flux calibration observing a standard star having CALSPEC data.
+import glob
+import os
+import numpy as np
+from astropy.io import fits, ascii
+from scipy import integrate
+import urllib
+
+
+# Dictionary of anticipated bright and dim CASLPEC standard star names and corresponding fits names
+calspec_names= {
+# bright standards
+'109 Vir': '109vir_stis_005.fits',
+'Vega': 'alpha_lyr_stis_011.fits',
+'Eta Uma': 'etauma_stis_008.fits',
+'Lam Lep': 'lamlep_stis_008.fits',
+'KSI2 CETI': 'ksi2ceti_stis_008.fits',
+# dim standards
+'TYC 4433-1800-1': '1808347_stiswfc_006.fits',
+'TYC 4205-1677-1': '1812095_stisnic_008.fits',
+'TYC 4212-455-1': '1757132_stiswfc_006.fits',
+'TYC 4209-1396-1': '1805292_stisnic_008.fits',
+'TYC 4413-304-1': 'p041c_stisnic_010.fits',
+'UCAC3 313-62260': 'kf08t3_stisnic_005.fits',
+'BPS BS 17447-0067': '1802271_stiswfcnic_006.fits',
+'TYC 4424-1286-1': '1732526_stisnic_009.fits',
+'GSC 02581-02323': 'p330e_stiswfcnic_007.fits',
+'TYC 4207-219-1': '1740346_stisnic_005.fits'
+}
+
+calspec_url = 'https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/'
+
+def get_calspec_file(star_name):
+    """
+    download the corresponding CALSPEC fits file and return the file path
+    
+    Args:
+        star_name (str): 
+    
+    Returns:
+        str: file path
+    """
+    if star_name not in calspec_names:
+        raise ValueError('{0} is not in list of anticipated standard stars {1}, please check naming'.format(star_name, calspec_names.keys()) )
+    fits_name = calspec_names.get(star_name)
+    # TODO: be flexible with the version of the calspec fits file, so essentially, the number in the name should not matter
+    fits_url = calspec_url + fits_name
+    try:
+        calspec_dir = os.path.join(os.path.dirname(__file__), "data", "calspec_data")
```

Comment:
because we can't assume we always have write access to the folder in which this file resides on a user's computer, we should download new files into the .corgidrp folder that stores the caldb and config files: `os.path.dirname(corgidrp.config_filepath)` (I need to update this so we can directly access the path to the .corgidrp folder instead of having to do dirname()). 

You are free to make a folder called calspec_data in ~/.corgidrp though!

Reply:
Ok, I did that